src/controller/replication/flow/copy.go

```diff
@@ -127,11 +145,13 @@ func (c *copyFlow) createTasks(ctx context.Context, srcResources, dstResources [
 
 		if _, err = c.taskMgr.Create(ctx, c.executionID, job, map[string]interface{}{
 			"operation":            "copy",
-			"resource_type":        string(resource.Type),
-			"source_resource":      getResourceName(resource),
-			"destination_resource": getResourceName(dstResources[i])}); err != nil {
+			"resource_type":        string(srcResource.Type),
+			"source_resource":      getResourceName(srcResource),
+			"destination_resource": getResourceName(dstResource)}); err != nil {
 			return err
 		}
+
+		taskCnt++
```

Comment:
Is it possible that even the task creation returns error, but the task still to be created? And in this situation, what will happen after calls MarkDone()?

Reply:
no, if create task error, task manager will ensure no task in database and jobservice.